plugins/net_plugin/net_plugin.cpp

```diff
@@ -1921,7 +1922,8 @@ namespace eosio {
    }
 
    // call with g_sync locked, called from conn's connection strand
-   void sync_manager::request_next_chunk( std::unique_lock<std::mutex> g_sync, const connection_ptr& conn ) {
+   void sync_manager::request_next_chunk( fc::mutex *, const connection_ptr& conn ) RELEASE(sync_mtx) {
+      fc::lock_guard g(sync_mtx, fc::adopt_lock);
```

Comment:
No, the method should require the mutex be held. Not acquire it.
Also please revert the other logic changes.

Reply:
It does require the mutex being help (`RELEASE(sync_mtx)`) verifies this. It doesn't acquire it.
I'll revert the other logic changes (even though I think it is cleaner as is now).

Reply:
I think we can change the method to not pass the mutex* and not adopt the lock. Just require the mutex to be locked.

Reply:
It is fine as is, I just thought we were going to just hold the lock during this private method call. Not `adopt_lock`.

Reply:
Oh, I understand! changing it.

Reply:
Done. You are right it is much better like that! Sorry I was so slow to understand what you meant.